Source/Core/PolygonGeometry.js

```diff
@@ -681,6 +681,10 @@ define([
         var polygons = results.polygons;
         var i;
 
+        if (polygonHierarchy.positions < 3) {
```

Comment:
actually, you'll need to check `outerPositions.length` (around line 684) instead, since duplicate positions are removed in `PolygonGeometryLibrary.polygonsFromHierarchy`